lms/djangoapps/courseware/views/views.py

```diff
@@ -937,14 +938,22 @@ def _get_cert_data(student, course, course_key, is_active, enrollment_mode):
         course = get_course_by_id(course_key)
         may_view_certificate = course.self_paced or course.may_certify()
 
-    show_message = all([
+    switches = certificates_waffle.waffle()
+    switches_enabled = (switches.is_enabled(certificates_waffle.SELF_PACED_ONLY) and
+                        switches.is_enabled(certificates_waffle.INSTRUCTOR_PACED_ONLY))
+    student_cert_generation_enabled = certs_api.cert_generation_enabled(course_key) if not switches_enabled else True
```

Comment:
I think this actually fixes a bug: https://openedx.atlassian.net/browse/EDUCATOR-1292

Reply:
It does? I don't see how it would affect anything about grading.

Reply:
Sorry, that was the wrong ticket, this is the right one: https://openedx.atlassian.net/browse/EDUCATOR-1281

Reply:
I'm not sure if that's a bug so much as expected functionality that isn't clearly stated? We talked about it last week and decided not to get rid of that "enable student-generated certificates" button, so this will still work the same way.